apps/gitness/src/components/stores/recent-pinned-nav-links.store.ts

```diff
@@ -0,0 +1,78 @@
+import { create, StateCreator } from 'zustand'
+import { persist } from 'zustand/middleware'
+
+import { NavbarItemType } from '@harnessio/ui/components'
+
+export interface NavState {
+  recent: NavbarItemType[]
+  pinned: NavbarItemType[]
+  setRecent: (route: NavbarItemType) => void
+  setPinned: (route: NavbarItemType, pin: boolean) => void
+  setNavLinks: (links: Partial<Pick<NavState, 'pinned' | 'recent'>>) => void
+}
+
+const navStateCreator: StateCreator<NavState> = set => ({
+  recent: [],
+  pinned: [],
+
+  setRecent: (route, remove = false) =>
+    set(state => {
+      let copyRecent = [...state.recent]
+
+      if (remove) {
+        copyRecent = copyRecent.filter(item => item.id !== route.id)
+      } else {
+        // Check if the route already exists in the history
+        if (state.recent.find(item => item.id === route.id)) return state
+
+        // Check if the route already exists in the pinned list
+        if (state.pinned.find(item => item.id === route.id)) return state
```

Comment:
could we use a Set to avoid iterating an array for finding duplicates?
not a major issue, since it'll be a small array anyway.. but could grow if we're adding all pages in the future

Reply:
sure